gio/src/initable.rs

```diff
@@ -0,0 +1,34 @@
+use crate::traits::InitableExt;
+use crate::Cancellable;
+use crate::Initable;
+use glib::object::IsA;
+use glib::object::IsClass;
+use glib::value::ToValue;
+use glib::Object;
+
+pub trait InitableExtManual: Sized + 'static {
+    /// Helper function to safely create and initialize an object implementing [Initable].
+    fn initable_new<P: IsA<Cancellable>>(
```

Comment:
You'd put that into an `impl Initiable` block, and then

```suggestion
    fn initable_new<T: IsA<Initable>, P: IsA<Cancellable>>(...) -> Result<T, Error>
```

Reply:
Or maybe not... mh!

Reply:
The reason I did it in an external trait (`InitableExtManual`) was to preserve the generation of the base interface from gir. I can move the whole Initable to manual if preferred, though.

Reply:
You can keep the autogenerated code and have a separate `impl Initiable` in `src/initable.rs`. That's not a problem :)